<?xml version="1.0"?>

<robot xmlns:xacro="http://www.ros.org/wiki/xacro" name="robot">

    <!-- <xacro:include filename="$(find neeeilit)/urdf/macros.xacro" /> -->

    <xacro:property name="joint_lower_limit" value="-${1.57}" />
	<xacro:property name="joint_upper_limit" value="${1.57}" />
	<xacro:property name="joint_effort" value="6.0" />
	<xacro:property name="joint_velocity" value="0.524" />

    <link name="base_link">
        <visual>
            <origin xyz="0.0 0.0 0.0" rpy="0.0 0.0 0.0"/>
            <geometry>
                <box size="1 1 1"/> 
            </geometry>
        </visual>
        <collision>
            <origin xyz="0.0 0.0 0" rpy="0.0 0.0 0.0"/>
            <geometry>
                <box size="1 1 1"/> 
            </geometry>
        </collision>
        <!-- <inertial>
            <mass value="1024"/>
            <inertia 
                ixx="170.667" ixy="0.0" ixz="0.0"
                iyy="170.667" iyz="0.0"
                izz="170.667"/>
        </inertial> -->
    </link>



    <link name="base_to_joint1">
        <visual>
            <origin rpy="0 0 0.0" xyz="0 0 0.1"/>
            <geometry>
                <cylinder radius="0.5" length="0.2"/>
            </geometry>
        </visual>
        <collision>
            <origin rpy="0 0 0" xyz="0 0 0.1"/>  
            <geometry>
                <cylinder radius="0.5" length="0.2"/>
            </geometry>
        </collision>
        <inertial>
            <mass value="1"/>
            <inertia 
                ixx="0.06" ixy="0.0" ixz="0.0" 
                iyy="0.06" iyz="0.0" 
                izz="0.25"/>
        </inertial>
    </link>

    <joint name="joint1" type="revolute">
        <origin rpy="0 0 0" xyz="0 0 0.5"/>
        <parent link="base_link"/>
        <child link="base_to_joint1"/>
        <axis xyz="0 0 1"/>
        <limit lower="-1.57" upper="1.57" effort="1000.0" velocity="0.5" />
    </joint>

    <transmission name="joint1_transmission">
        <type>transmission_interface/SimpleTransmission</type>
        <joint name="joint1">
            <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
        </joint>
        <actuator name="joint_motor1">
            <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
            <mechanicalReduction>1</mechanicalReduction>
        </actuator>
    </transmission>

    

    <link name="joint1_to_joint2">
        <visual>
            <origin rpy="0 0 0.0" xyz="0 0 0.4"/>
            <geometry>
                <cylinder radius="0.15" length="0.80"/>
            </geometry>
        </visual>
        <collision>
            <origin rpy="0 0 0" xyz="0 0 0.4"/>  
            <geometry>
                <cylinder radius="0.15" length="0.8"/>
            </geometry>
        </collision>
        <inertial>
            <mass value="57.906"/>
            <inertia 
                ixx="12.679" ixy="0.0" ixz="0.0" 
                iyy="12.679" iyz="0.0" 
                izz="0.651"/>
        </inertial>
    </link>

    <joint name="joint2" type="revolute">
        <origin rpy="0 0 0" xyz="0 0 0.2"/>
        <parent link="base_to_joint1"/>
        <child link="joint1_to_joint2"/>
        <axis xyz="0 1 0"/>
        <limit lower="-1.57" upper="1.57" effort="1000.0" velocity="0.5" />
    </joint>

    <transmission name="joint2_transmission">
        <type>transmission_interface/SimpleTransmission</type>
        <joint name="joint2">
            <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
        </joint>
        <actuator name="joint_motor2">
            <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
            <mechanicalReduction>1</mechanicalReduction>
        </actuator>
    </transmission>



    <link name="joint2_to_joint3">
        <visual>
            <origin rpy="0 0 0.0" xyz="0 0 0.15"/>
            <geometry>
                <cylinder radius="0.125" length="0.30"/>
            </geometry>
        </visual>
        <collision>
            <origin rpy="0 0 0" xyz="0 0 0.15"/>  
            <geometry>
                <cylinder radius="0.125" length="0.30"/>
            </geometry>
        </collision>
        <inertial>
            <mass value="57.906"/>
            <inertia 
                ixx="12.679" ixy="0.0" ixz="0.0" 
                iyy="12.679" iyz="0.0" 
                izz="0.651"/>
        </inertial>
    </link>

    <joint name="joint3" type="revolute">
        <origin rpy="0 0 0" xyz="0 0 0.8"/>
        <parent link="joint1_to_joint2"/>
        <child link="joint2_to_joint3"/>
        <axis xyz="0 1 0"/>
        <limit lower="-1.57" upper="1.57" effort="1000.0" velocity="0.5" />
    </joint>

    <transmission name="joint3_transmission">
        <type>transmission_interface/SimpleTransmission</type>
        <joint name="joint3">
            <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
        </joint>
        <actuator name="joint_motor3">
            <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
            <mechanicalReduction>1</mechanicalReduction>
        </actuator>

    </transmission>
    


    <link name="joint3_to_joint4">
        <visual>
            <origin rpy="0 0 0.0" xyz="0 0 0.15"/>
            <geometry>
                <cylinder radius="0.075" length="0.30"/>
            </geometry>
        </visual>
        <collision>
            <origin rpy="0 0 0" xyz="0 0 0.15"/>  
            <geometry>
                <cylinder radius="0.075" length="0.30"/>
            </geometry>
        </collision>
        <inertial>
            <mass value="57.906"/>
            <inertia 
                ixx="12.679" ixy="0.0" ixz="0.0" 
                iyy="12.679" iyz="0.0" 
                izz="0.651"/>
        </inertial>
    </link>

    <joint name="joint4" type="revolute">
        <origin rpy="0 0 0" xyz="0 0 0.3"/>
        <parent link="joint2_to_joint3"/>
        <child link="joint3_to_joint4"/>
        <axis xyz="0 1 0"/>
        <limit lower="-1.57" upper="1.57" effort="1000.0" velocity="0.5" />
    </joint>
    
    <transmission name="joint4_transmission">
        <type>transmission_interface/SimpleTransmission</type>
        <joint name="joint4">
            <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
        </joint>
        <actuator name="joint_motor4">
            <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
            <mechanicalReduction>1</mechanicalReduction>
        </actuator>
    </transmission>

    

    <!-- Add ROS plugin so we can interact with our robot -->
    <gazebo>
        <plugin name="gazebo_ros_control" filename="libgazebo_ros_control.so">
            <robotNamespace>/</robotNamespace>
        </plugin>
    </gazebo>

</robot>